src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -56,14 +58,41 @@ public static class TaskExtensions {
         }
 
         /// <summary>
-        /// Suppresses warnings about unawaited tasks and ensures that unhandled
-        /// errors will cause the process to terminate.
+        /// Rethrows task exceptions back to the callers synchronization context if it is possible
         /// </summary>
         /// <remarks>
         /// <see cref="OperationCanceledException"/> is always ignored.
         /// </remarks>
-        public static async void DoNotWait(this Task task) {
-            await task.SilenceException<OperationCanceledException>();
+        public static void DoNotWait(this Task task) {
```

Comment:
@zooba, you might want to take a look, since this was originally a piece of code borrowed from PTVS, and it might be good to merge the changes back at some point.